recipes_source/torch_compiler_set_stance_tutorial.py

```diff
@@ -0,0 +1,232 @@
+        return x + 1
+
+
+try:
+    baz(inp)
+except Exception as e:
+    print(e)
+
+
+@torch.compiler.set_stance("force_eager")
+def inner(x):
+    return x + 1
+
+
+@torch.compile
+def outer(x):
+    # error!
+    return inner(x)
+
+
+try:
+    outer(inp)
+except Exception as e:
+    print(e)
+
+######################################################################
+# Other stances include:
+#  - ``"default"``: The default stance, used for normal compilation.
+#  - ``"eager_on_recompile"``: Run code eagerly when a recompile is necessary. If there is cached compiled code valid for the input, it will still be used.
+#  - ``"fail_on_recompile"``: Raise an error when recompiling a function.
+#
+# See the ``torch.compiler.set_stance`` `doc page <https://pytorch.org/docs/main/generated/torch.compiler.set_stance.html#torch.compiler.set_stance>`__
+# for more stances and options. More stances/options may also be added in the future.
+
+######################################################################
+# Examples
+# --------
+
+######################################################################
+# Preventing recompilation
+# ========================
+#
+# Some models do not expect any recompilations - for example, you may always inputs to be the same shape.
+# Since recompilations may be expensive, we may wish to error out when we attempt to recompile so we can detect and fix recompilation cases.
+# The ``"fail_on_recompilation"`` stance can be used for this.
+
+
+@torch.compile
+def my_big_model(x):
+    return torch.relu(x)
+
+
+# first compilation
+my_big_model(torch.randn(3))
+
+with torch.compiler.set_stance("fail_on_recompile"):
+    my_big_model(torch.randn(3))  # no recompilation - OK
+    try:
+        my_big_model(torch.randn(4))  # recompilation - error
+    except Exception as e:
+        print(e)
+
+######################################################################
+# If erroring out is too disruptive, we can use ``"eager_on_recompile"`` instead,
+# which will cause ``torch.compile`` to fall back to eager instead of erroring out.
+# This may be useful if we don't expect recompilations to happen frequently, but
+# when one is required, we'd rather pay the cost of running eagerly over the cost of recompilation.
+
+
+@torch.compile
+def my_huge_model(x):
+    if torch.compiler.is_compiling():
+        return x + 1
+    else:
+        return x - 1
+
+
+# first compilation
+print(my_huge_model(torch.zeros(3)))  # 1
+
+with torch.compiler.set_stance("eager_on_recompile"):
+    print(my_huge_model(torch.zeros(3)))  # 1
+    print(my_huge_model(torch.zeros(4)))  # -1
+    print(my_huge_model(torch.zeros(3)))  # 1
+
+
+######################################################################
+# Measuring performance gains
+# ===========================
+#
+# ``torch.compiler.set_stance`` can be used to compare eager vs. compiled performance
+# without having to define a separate eager model.
+
```

Comment:
did you mean to have a piece of code here?

Reply:
Below comment (`Returns the result...`) is part of code.